src/DataCollector.tpp

```diff
@@ -0,0 +1,73 @@
+#ifndef DATACOLLECTOR_TPP
+#define DATACOLLECTOR_TPP
+
+#include "DataCollector.h"
+#include <SensorData.h>
+
+DC_TP DC_IMPL::DataCollector(uint32_t id, size_t numSensors, Sensor** sensors, const bool debug)
+{
+    this->id = id;
+    this->numSensors = numSensors;
+    this->sensors = sensors;
+    this->debug = debug;
+}
+
+DC_TP void DC_IMPL::begin(FlexCAN_T4<_bus, _rxSize, _txSize>* can)
+{
+    this->can = can;
+    health = HEALTHY;
+}
+
+DC_TP void DC_IMPL::read(Sensor* sensor)
```

Comment:
One other thing. We didn't chat about this but after reviewing some judge feedback I think having all critical sensors being sent on the motorCAN instead of the commsCAN is smart. 

I can hear them ask if we isolated the motor and critical function messages why didn't we isolate coms for the sensors that control those critical functions? 

Having it use is critical to decide what line to send it on would be a good add imo.